cpp/util/ETSWrapper.h

```diff
@@ -0,0 +1,106 @@
+#ifndef ETSWRAPPER_H
+#define ETSWRAPPER_H
+
+#include <tbb/tbb.h>
+
+namespace freud { namespace util {
+
+//! Wrapper classfor enumerable_thread_specific<T*>
+/*! It is expected that default value for T is 0.
+ */
+template<typename T> class ETSArrayWrapper
```

Comment:
I would prefer a name like `ThreadStorage` that is clearer to new readers. I don't think the abbreviation ETS will make sense unless a reader knows TBB well.